utils/cli/multisend.go

```diff
@@ -0,0 +1,419 @@
+				if err != nil {
+					return err
+				}
+			}
+
+			if progress.SequenceHotWallet == 0 {
+				progress.SequenceHotWallet = progress.SequenceOrigin
+				if useHotWallet {
+					progress.SequenceHotWallet, err = getSequence(clientCtxHotWallet.FromAddress.String())
+					if err != nil {
+						return err
+					}
+				}
+			}
+
+			output := []banktypes.Output{}
+			totalAmount := sdk.Coins{}
+			records = records[progress.Index:]
+			for i, record := range records {
+				coins, err := sdk.ParseCoinsNormalized(record[1])
+				if err != nil {
+					fmt.Printf("failed sending records from %d to %d\n", progress.Index, i)
+					fmt.Printf("please run again with: multi-send [file.csv] %d\n", progress.Index)
+					return err
+				}
+
+				if coins.IsZero() {
+					fmt.Printf("failed sending records from %d to %d\n", progress.Index, i)
+					fmt.Printf("please run again with: multi-send [file.csv] %d\n", progress.Index)
+					return fmt.Errorf("must send positive amount")
+				}
+				toAddr, err := sdk.AccAddressFromBech32(record[0])
+				if err != nil {
+					fmt.Printf("failed sending records from %d to %d\n", progress.Index, i)
+					fmt.Printf("please run again with: multi-send [file.csv] %d\n", progress.Index)
+					return err
+				}
+				output = append(output, banktypes.NewOutput(toAddr, coins))
+				totalAmount = totalAmount.Add(coins...)
+
+				if (i+1)%MAX_ADDRESSES == 0 || (i+1) == len(records) {
+					if useHotWallet {
+						if progress.Progress == PRG_ready {
+							msg := banktypes.NewMsgSend(clientCtxOrigin.FromAddress, clientCtxHotWallet.FromAddress, totalAmount.Add(fees...))
+							err := tx.GenerateOrBroadcastTxCLI(clientCtxOrigin, cmd.Flags(), msg)
+							if err != nil {
+								fmt.Printf("failed sending records from %d to %d\n", progress.Index, i)
+								fmt.Printf("please run again with: multi-send [file.csv] %d\n", progress.Index)
+								return err
+							}
+							progress.Progress = PRG_bank_send
+							progress.SequenceOrigin++
+							saveProgress(progress, progressFile)
+						}
+						if progress.Progress == PRG_bank_send {
+							err = waitSequenceChange(clientCtxOrigin.FromAddress.String(), progress.SequenceOrigin)
+							if err != nil {
+								fmt.Printf("failed sending records from %d to %d\n", progress.Index, i)
+								fmt.Printf("please run again with: multi-send [file.csv] %d\n", progress.Index)
+								return err
+							}
+							progress.Progress = PRG_bank_send_verified
+							saveProgress(progress, progressFile)
+						}
+					}
+
+					if progress.Progress == PRG_bank_send_verified || progress.Progress == PRG_ready {
+						msg := banktypes.NewMsgMultiSend([]banktypes.Input{banktypes.NewInput(clientCtxHotWallet.FromAddress, totalAmount)}, output)
+						fmt.Printf("sending records from %d to %d, total tokens %s\n", progress.Index, i, output[0].String())
+						err := tx.GenerateOrBroadcastTxCLI(clientCtxHotWallet, cmd.Flags(), msg)
```

Comment:
if there was an error in this broadcast and we run again it will send the tx again on the same indexes, it does not check the sequence number of the hot wallet prior to make sure it didn't advance, very risky need to fix

Reply:
done